[gerteck] iP

gerteck opened the pull request:
## Ellie Chatbot
> "It's not enough to be busy. The question is, what are we busy about?" - Henry David Thoreau ([source](https://ca.indeed.com/career-advice/career-development/quotes-on-time-management))


Ellie frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://github.com/gerteck/ip/releases/tag/untagged-2a1ad244f5c86e673323).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉
And it is **FREE!**

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

gerteck committed:
L0: Rename, Greet, Exit

gerteck committed:
L1: Echo

gerteck committed:
L2: Add, List

gerteck committed:
A-Classes

gerteck committed:
L3: Add Mark as Done

gerteck committed:
Add Task subtypes D, T, E.

gerteck committed:
L4: Add support for three task types

gerteck committed:
Added Automated Text UI Testing

gerteck committed:
Add Exception Classes

gerteck committed:
Add Delete Functionality

gerteck committed:
Refactor files to Add Storage Class

gerteck committed:
Add ignore clause for /data

gerteck committed:
L7: Add Save Functionality

Added save and load functionality to the Storage class.
Tracker calls storage to save and load at appropriate points

gerteck committed:
Merge branch 'branch-Level-7'

gerteck committed:
L8: Add Date Parsing

gerteck committed:
Merge branch 'branch-Level-8'

gerteck committed:
Add Ui Class

gerteck committed:
Refactor Classes into packages

gerteck committed:
Add Command Classes

gerteck committed:
Refactor code to incorporate more OOP

Added Command classes to represent different commands.
Refactored Functionality into Parser, UI, Storage, Tasklist classes.

gerteck committed:
Merge branch 'A-MoreOOP-Packages'

gerteck committed:
Sync with Remote Repo for Pull

gerteck committed:
Merge branch 'add-gradle-support'

gerteck committed:
Set up Gradle

gerteck committed:
Merge branch 'A-Gradle'

gerteck committed:
Add Junit Tests

Added Junit Tests for Parser, TaskList Classes

gerteck committed:
Merge branch 'A-JUnit'

gerteck committed:
Configure Shadow Plugin

gerteck committed:
A-JavaDoc: Add JavaDoc comments to code

Added header comments to all non-trivial methods and classes

gerteck committed:
L9: Add Find Functionality

gerteck committed:
Merge branch 'branch-A-JavaDoc'

gerteck committed:
Merge branch 'branch-Level-9'

Resolved Conflict in TaskList.java

gerteck committed:
Add Checkstyle XML files

gerteck committed:
Refactor code to conform to Checkstyle

gerteck committed:
Merge branch 'A-CheckStyle'

gerteck committed:
Add JavaFX Elements

Link to Ellie

gerteck committed:
Refactor using FXML

gerteck committed:
Merge branch 'branch-Level-10'

sdevih commented on src/main/java/ellie/command/FindCommand.java:
```diff
@@ -24,9 +24,9 @@ public FindCommand(String keyword) {
-    public void run(TaskList tasklist) {
-        tasklist.searchTask(keyword);
-        return;
+    public String runAndReturnResponse(TaskList tasklist) {
+        String response = tasklist.searchTask(keyword);
+        return response;
```
I like how you adhered to the conventions of naming methods which makes it clear what the method does.

sdevih commented on src/main/java/ellie/command/DeleteCommand.java:
```diff
@@ -26,9 +26,9 @@ public DeleteCommand(int index) {
-    public void run(TaskList tasklist) {
-        tasklist.deleteTaskIndex(index);
-        return;
+    public String runAndReturnResponse(TaskList tasklist) {
+        String response = tasklist.deleteTaskIndex(index);
+        return response;
```
Since this is an overridden method inherited from the parent class, you may consider using `@inheritDoc` tag, which allows you to reuse the header comment from the parent method but with further modifications.

sdevih commented on src/main/java/ellie/Ui.java:
```diff
@@ -6,8 +6,8 @@
 
-    final static String horizontalLine = "____________________________________________________________";
-    final static String logoEllie =
+    final static String HORIZONTAL_LINE = "____________________________________________________________";
+    final static String LOGO_ELLIE =
```
Good job on using uppercase letters for the constants!

sdevih commented on src/main/java/ellie/Ui.java:
```diff
@@ -26,20 +26,23 @@ public Ui() {
-        System.out.println("Hello! I'm Ellie, your CS2103T chat bot! I help by tracking your tasks!");
-        System.out.println(logoEllie + "\n" + horizontalLine);
-        System.out.println("What can I do for you? Type 'help' to see available commands! \n");
+    public static String hello() {
+        String helloMessage = "";
+        helloMessage += "Hello! I'm Ellie, your CS2103T chat bot! I help by tracking your tasks!\n";
+        helloMessage += LOGO_ELLIE + "\n" + HORIZONTAL_LINE + "\n";
+        helloMessage += "What can I do for you? Type 'help' to see available commands! \n";
+        return helloMessage;
```
Since names representing methods must be verbs and written in camelCase, you may consider changing the method name here to `showHelloMessage()` as a method name.

sdevih reviewed (commented):
Looks great overall! I have commented on just some minor fixes. Keep up the great work!

gerteck committed:
Refactor MainWindow

Increased Opacity for User and Ellie Profile Images in DialogBox.fxml. Added starting welcome message for chatbot in MainWindow.java

gerteck committed:
A-Assertions: Add Assertions

Add assertions to Parser and Storage class.

For Parser class, add assertion that command should be properly parsed.
For Storage class, add assertions that filePath and directory should not
be null, that save file exists before saving and save file contents are
in the correct format.

gerteck committed:
Merge pull request #1 from gerteck/branch-A-Assertions

A-Assertions: Add Assertions

gerteck committed:
Refactor Code Quality

Rename methods Hello() to showHelloMessage() and Goodbye() to
showGoodbyeMessage() to conform to better code quality standards.

gerteck committed:
Refactor Code Quality

Changed method name for Ui methods to better reflect methods. Added
@inheritDoc for DeleteCommand.java method.

gerteck committed:
Merge pull request #2 from gerteck/branch-A-CodeQuality

Improve code quality

Critically examined code and refactored to improve the code quality where necessary.

gerteck committed:
A-CI: Set up Continuous Integration

gerteck committed:
Merge branch 'master' of https://github.com/gerteck/ip

gerteck committed:
Add Extension C-Help

gerteck committed:
Refactor Code Quality to pass Checkstyle

Made edits to variable naming, import ordering, javadocs etc. to pass
CheckStyle Tests

gerteck committed:
Major Overhaul of GUI

Changed Label to Text, enabling full dialogText to be shown.
Added background colors to GUI and relevant Nodes.
Added relevant height and width properties for resizing GUI.

gerteck committed:
Replace Images

Replaced Image for Ellie & User.

gerteck committed:
Merge branch 'branch-A-BetterGui'

gerteck committed:
Add Cosmetic Changes

Added Application name & Icon.
Also refactored code to conform to checkstyle.

gerteck committed:
Merge branch 'branch-A-Personality'

gerteck committed:
Refactor Code

Renamed Application to flesh out chatbot name.
Fixed GUI background color bug.

gerteck committed:
Edit UI response

Added a cosmetic line to chatbot response.

gerteck committed:
Refactor Code Quality

Apply SLAP (Single Level of Abstraction Principle).
Replaced Magic Literals.
Refactored long methods by extracting portions into helper functions.

gerteck committed:
Merge branch 'branch-A-CodeQuality'

gerteck committed:
Add User Guide

gerteck committed:
Add Screenshots for User Guide

gerteck committed:
Refactor Code

Refactor code quality and fix typos.

gerteck committed:
Add cosmetic changes

gerteck committed:
Add minor changes

Added some comments to improve readability of code.

gerteck committed:
Add ReadMe for Github Repository
